tests/functional/serving/test_storage_resource.py

```diff
@@ -0,0 +1,436 @@
+        if os.path.isfile(path):
+            os.unlink(path)
+
+
+def _random_string(length: int):
+    """Generate a random string with given length. Used to generate multipart form data for /load request"""
+    domain = string.ascii_lowercase + string.digits
+    return "".join(random.choice(domain) for _ in range(length))
+
+
+def _gen_multipart_form_form_load(file: str, resolve: str, name="", version=""):
+    """Generate multipart form body and headers for /load request with uploaded file.
+    Help (2022/11/08):
+    https://stackoverflow.com/a/45703104
+    https://stackoverflow.com/a/23517227
+    """
+    import io
+
+    boundary = "----MultiPartFormBoundary" + _random_string(16)
+    in_boundary = "--" + boundary
+    out_boundary = in_boundary + "--"
+
+    buff = io.BytesIO()
+
+    buff.write(in_boundary.encode())
+    buff.write(b"\r\n")
+    buff.write(b'Content-Disposition: form-data; name="file"; filename="load.pkl"')
+    buff.write(b"\r\n")
+    buff.write(b"Content-Type: application/octet-stream")
+    buff.write(b"\r\n")
+    buff.write(b"\r\n")
+    with open(file, "rb") as data:
+        buff.write(data.read())
+    buff.write(b"\r\n")
+
+    buff.write(in_boundary.encode())
+    buff.write(b"\r\n")
+    buff.write(b'Content-Disposition: form-data; name="resolve"')
+    buff.write(b"\r\n")
+    buff.write(b"\r\n")
+    buff.write(resolve.encode())
+    buff.write(b"\r\n")
+
+    buff.write(in_boundary.encode())
+    buff.write(b"\r\n")
+    buff.write(b'Content-Disposition: form-data; name="name"')
+    buff.write(b"\r\n")
+    buff.write(b"\r\n")
+    buff.write(name.encode())
+    buff.write(b"\r\n")
+
+    buff.write(in_boundary.encode())
+    buff.write(b"\r\n")
+    buff.write(b'Content-Disposition: form-data; name="version"')
+    buff.write(b"\r\n")
+    buff.write(b"\r\n")
+    buff.write(version.encode())
+    buff.write(b"\r\n")
+
+    buff.write(out_boundary.encode())
+    buff.write(b"\r\n")
+
+    headers = {
+        "Content-Type": f"multipart/form-data; boundary={boundary}",
+        "Content-Length": str(buff.tell()),
+    }
+
+    return buff.getvalue(), headers
+
+
+def test_dump_all(client, ephemeral_storage):
+    """Test simple call to /dump"""
+    _load_data(ephemeral_storage)
+    response = client.simulate_get("/dump")
+    host = _gen_host_file()
+    try:
+        with open(host, "wb") as file:
+            file.write(response.content)
+        dumped_db = PickledDB(host)
+        with dumped_db.locked_database(write=False) as internal_db:
+            collections = set(internal_db._db.keys())
+        assert collections == {"experiments", "algo", "trials", "benchmarks"}
+        assert len(dumped_db.read("experiments")) == 3
+        assert len(dumped_db.read("algo")) == 3
+        assert len(dumped_db.read("trials")) == 24
+        assert len(dumped_db.read("benchmarks")) == 0
+    finally:
+        _clean_dump(host)
+
+
+def test_dump_one_experiment(client, ephemeral_storage):
```

Comment:
Since the interface is very similar it may be possible to refactor the tests so that they reuse the same code blocks for data validation.

Reply:
Done!